### PR TITLE
Add overlay and walk strategy loading from a function

### PR DESCRIPTION
**Pull request by drew2a**
This PR:

Adds the ability to load overlay and walk strategy from "function class representation" to have both explicit imports and lazy loading:

``` python
def GigaChannelCommunity():
    from tribler_core.modules.metadata_store.community.gigachannel_community import GigaChannelCommunity
    return GigaChannelCommunity

 ...

@overlay(GigaChannelCommunity)
class GigaChannelCommunityLauncher(IPv8CommunityLauncher):
    pass
```
 
Reference: https://github.com/Tribler/tribler/pull/5746

### CODE REVIEW

**Comment by qstokkink** [comment]
ok to test

File: ipv8/loader.py [hunk 1]
```diff
@@ -118,6 +119,15 @@ class A(CommunityLauncher):
...
            ...
 
+        def MyCommunityClass():
+            from my_module.some_submodule import MyCommunityClass
+            return MyCommunityClass
+           ...
+
+        @overlay(MyCommunityClass)
+        class A(CommunityLauncher):
+            pass
```
Comment (qstokkink) [inline ipv8/loader.py:129]: This new example redefines `MyCommunityClass` and uses `...` differently. This issue is repeated at line 173. Please use something close (or equal) to:

```suggestion
        def my_community_class():
            from my_module.some_submodule import MyCommunityClass
            return MyCommunityClass

        @overlay(my_community_class)
        class A(CommunityLauncher):
            ...
```

File: ipv8/util.py [hunk 2]
```diff
@@ -38,3 +40,8 @@ async def call_async(*args, **kwargs):
 
 def strip_sha1_padding(s: bytes):
     return s[12:] if s.startswith(b'SHA-1\x00\x00\x00\x00\x00\x00\x00') else s
+
+
+def get_class(class_or_function):
```
Comment (qstokkink) [inline ipv8/util.py:45]: This function is not used outside of `loader.py`, please relocate it to that file.

**Review by qstokkink — CHANGES_REQUESTED** [review]
Thanks for your contribution. Please fix the three issues with this PR so it can be merged.

File: ipv8/util.py (hunk 2 above)
Comment (drew2a) [reply ipv8/util.py]: Relocated :)

File: ipv8/loader.py (hunk 1 above)
Comment (drew2a) [reply ipv8/loader.py]: 👌

File: ipv8/loader.py [hunk 3]
```diff
@@ -160,6 +181,15 @@ class A(CommunityLauncher):
...
            ...
 
+       def my_strategy_class():
+            from my_module.some_submodule import MyStrategyClass
+            return MyStrategyClass
+           ...
+
+        @walk_strategy(my_strategy_class)
+        class A(CommunityLauncher):
+            pass
```
Comment (qstokkink) [inline ipv8/loader.py:191]: Looking good 👍 , please don't forget the second half of the request for changes: consistent use of the `...`:

```suggestion
       def my_strategy_class():
            from my_module.some_submodule import MyStrategyClass
            return MyStrategyClass

        @walk_strategy(my_strategy_class)
        class A(CommunityLauncher):
            ...
```

After that, please squash your commits (please contact me if you need help and I'll give you the appropriate git commands - this should take about 30 seconds).

Then, finally, this should be completely done.

**Review by qstokkink — CHANGES_REQUESTED** [review]

**Comment by drew2a** [comment]
@qstokkink looks good now)

File: ipv8/loader.py [hunk 4]
```diff
@@ -118,6 +130,15 @@ class A(CommunityLauncher):
...
            ...
 
+        def my_community_class():
+            from my_module.some_submodule import MyCommunityClass
+            return MyCommunityClass
+
+        @overlay(my_community_class)
+        class A(CommunityLauncher):
+            pass
+           ...
```
Comment (qstokkink) [inline ipv8/loader.py:140]: The issue at line 190 was resolved, please make sure to tackle this one too.

**Review by qstokkink — APPROVED** [review]
Beautiful, thanks for addressing all the change requests 👍

**Pull request merged**